R-proj/R/volume.R

```diff
@@ -1,217 +1,196 @@
   } else {
-    if (Zono) {
-      print('No Zonotope defined from input!')
-    } else if (Vpoly) {
-      print('No V-polytope defined from input!')
-    } else {
-      print('No H-polytope defined from input!')
-    }
-    return(-1)
+    stop("Not a known polytope representation.")
   }
-  A = matrix(cbind(b,A), ncol = dim(A)[2] + 1)
-  A = matrix(rbind(r,A), ncol = dim(A)[2])
   
-  # set a too large vector for chebychev ball if it is not given as input
-  Cheb_ball = rep(0, dim(A)[2] + 5)
-  if (!is.null(Inputs$Chebychev)) {
-    Cheb_ball = Inputs$Chebychev
-  }
-  
-  # set flag for CV algorithm
-  annealing = FALSE
-  if (!is.null(Inputs$CG)) {
-    annealing = Inputs$CG
-  }
-  
-  # set flag for verbose mode
-  verbose = FALSE
-  if (!is.null(Inputs$verbose)) {
-    verbose = Inputs$verbose
-  }
-  
-  # set flag for Coordinate or Random Directions HnR
-  coordinate = TRUE
-  if (!is.null(Inputs$coordinate)) {
-    coordinate = Inputs$coordinate
+  if (!intersection) {
+    Mat = P$get_mat()
+    dimension = dim(Mat)[2] - 1
+  } else {
+    dimension = dim(V1)[2] - 1
   }
   
-  # set flag for rounding
-  rounding = FALSE
-  if (!is.null(Inputs$rounding)) {
-    rounding = Inputs$rounding
+  CG = FALSE
+  win_len = 0
+  C = 0
+  N = 0
+  ratio = 0
+  frac = 0
+  if (!missing(Algo)) {
+    if(!is.null(Algo$CG)) {
+      if(Algo$CG) {
+        CG = TRUE
+        
+        win_len=4*(dimension^2)+500
+        if(!is.null(Algo$window_len)){
+          win_len=Algo$window_len
+        }
+        C=2
+        if(!is.null(Algo$C)){
+          C=Algo$C
+        }
+        ratio=1-1/dimension
+        if(!is.null(Algo$ratio)){
+          ratio=Algo$ratio
+        }
+        N=500*C+(dimension^2)/2
+        if(!is.null(Algo$N)){
+          N=Algo$N
+        }
+        frac=0.1
+        if(!is.null(Algo$frac)){
+          frac=Algo$frac
+        }
+      } else if (!is.null(Algo$SOB)) {
+        if (!Algo$SOB) {
+          stop("You have to choose between two Algorithms!")
+        }
+      } else {
+        warning("CG is false and no flag for SOB. The latest algorithm will be used.")
+      }
+    }
   }
   
   # set the number of steps for the random walk
-  if (!is.null(Inputs$walk_length)) {
-    W=Inputs$walk_length
+  if (!missing(walk_length)) {
+    W = walk_length
   } else {
-    if (annealing) {
+    if (CG) {
       W = 1
     }else{
-      W = 10 + floor( dim(A)[2]/10 )
+      W = 10 + floor( dimension / 10 )
     }
   }
   
   # set the requested error
-  if (!is.null(Inputs$error)) {
-    e = Inputs$error
+  if (!missing(error)) {
+    e = error
   } else {
-    if (annealing) {
+    if (CG) {
       e = 0.2
     } else {
       e = 1
     }
   }
-  dimension = dim(A)[2] - 1
   
-  # [CV] initialization
-  win_len = 4 * ( dimension ^ 2 ) + 500
-  if (!is.null(Inputs$win_len)) {
-    win_len = Inputs$win_len
+  # set a too large vector for chebychev ball if it is not given as input
+  InnerB = rep(0, dimension + 5)
```

Comment:
not a good practice to put magic numbers (i.e. 5). why not computing inner ball here if missing and avoid tricks?

Reply:
volume.R removed